Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs

```diff
@@ -810,6 +817,39 @@ public void HandleUserConfirmationAsyncAlwaysSendWhileDisabled()
             Assert.IsFalse(alwaysSendValue);
         }
 
+        [TestMethod]
+        public void TestTrackErrorValidParameters()
+        {
+            var mockHttpNetworkAdapter = Mock.Of<IHttpNetworkAdapter>();
+            var storage = new Storage.Storage(new StorageAdapter("test.db"));
+            var ingestion = new IngestionHttp(mockHttpNetworkAdapter);
+            var channelGroup = new ChannelGroup(ingestion, storage, "app secret");
+            Crashes.Instance.OnChannelGroupReady(channelGroup, "app secret");
+            var semaphore = new SemaphoreSlim(0);
+            HandledErrorLog actualLog = null;
+            Mock.Get(mockHttpNetworkAdapter).Setup(adapter => adapter.SendAsync(
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<IDictionary<string, string>>(),
+                It.IsAny<string>(),
+                It.IsAny<CancellationToken>()))
+                .Callback((string uri, string method, IDictionary<string, string> headers, string content, CancellationToken cancellation) =>
+                {
+                    actualLog = JsonConvert.DeserializeObject<LogContainer>(content, LogSerializer.SerializationSettings).Logs.Single() as HandledErrorLog;
+                    semaphore.Release();
+                });
+
+            var exception = new System.Exception("Something went wrong.");
+            var properties = new Dictionary<string, string> { { "k1", "v1" }, { "p2", "v2"} };
+            Crashes.TrackError(exception, properties);
+
+            // Wait until the http layer sends the log.
+            semaphore.Wait();
```

Comment:
In case the test fails, it will hang.
Though not a big fan of timeouts (since its like sleeping in a way), I would put a time limit here.
Anyway wait is like sleep infinite, so finite is still better than infinite.